Code/Framework/AzCore/AzCore/Memory/AllocationRecords.h

```diff
@@ -120,10 +120,9 @@ namespace AZ
         */
         class AllocationRecords
         {
-            friend class MemoryDriller;
+         public:
             AZ_CLASS_ALLOCATOR(AllocationRecords, OSAllocator, 0);
```

Comment:
Is this safe to use an Allocator?

Reply:
My understanding is that the OS Allocator does not record allocations, only totals, and uses direct OS allocation calls (ie, real malloc/free)

Reply:
yes, just because it is using the OSAllocator which doesnt set AllocationRecords... however, I agree, this overly sensitive
The issue may have arised from NewAndDelete.inl, we may have overwritten global new/delete, so we cannot rely on them either. 
I want to eventually remove this and rely on just logging the allocation/deallocation and doing the analysis out-of-process. Removing the need to store this state.